hack/verify/lint.sh

```diff
@@ -13,26 +13,19 @@
 # See the License for the specific language governing permissions and
 # limitations under the License.
 
-# CI script to run staticcheck
-set -o errexit
-set -o nounset
-set -o pipefail
+# script to run linters
+set -o errexit -o nounset -o pipefail
 
 # cd to the repo root
-REPO_ROOT=$(git rev-parse --show-toplevel)
+REPO_ROOT="$(cd "$(dirname "${BASH_SOURCE[0]}")/../.." && pwd -P)"
 cd "${REPO_ROOT}"
 
-# enable modules and the proxy cache
-export GO111MODULE="on"
-GOPROXY="${GOPROXY:-https://proxy.golang.org}"
-export GOPROXY
+# build golangci-lint
+SOURCE_DIR="${REPO_ROOT}/hack/tools" hack/go_container.sh \
+  go build -o /out/golangci-lint github.com/golangci/golangci-lint/cmd/golangci-lint
 
-# build staticcheck
-BINDIR="${REPO_ROOT}/bin"
-# use the tools module
-cd "hack/tools"
-go build -o "${BINDIR}/staticcheck" honnef.co/go/tools/cmd/staticcheck
-# go back to the root
-cd "${REPO_ROOT}"
-
-"${BINDIR}/staticcheck" ./pkg/... ./cmd/... .
+# run golangci-lint
+GO111MODULE=on bin/golangci-lint \
+  --enable=golint --enable=vet --enable=gofmt \
+  --enable=misspell \
+  run ./pkg/... ./cmd/... .
```

Comment:
this includes staticcheck and other lints by default, we're explicitly enabling a few that are not on by default:
- standard go lints (golint, vet, gofmt)
- misspell

Reply:
iterative (hot) run:
```
[bentheelder@bentheelder-macbookpro:~/go/src/sigs.k8s.io/kind·2019-08-22T21:38:14-0700·shell-nits@245ca10]
$ time hack/verify/lint.sh 

real    0m7.842s
user    0m23.441s
sys     0m4.258s
```

much faster than running these individually.

I'm exploring the additional lints we can enable, but may leave that for a follow up.

Reply:
that's progress

Reply:
it's more like a minute on a cold run, but that's not bad considering how much static analysis we're getting. this already caught some bugs we hadn't before, just with the default checks.